scenarios.md

```diff
@@ -0,0 +1,199 @@
+# Notary Signing - Scenarios
+
+As containers and cloud native artifacts become the common unit of deployment, users want to know the artifacts in their environments are authentic and un-tampered. 
+
+These Notary v2 scenarios define end-to-end scenarios for signing artifacts, storing and moving them between OCI compliant registries, and validating them by various artifact hosts and tooling. 
+
+## OCI Images & Artifacts
+
+The [OCI TOB][oci-tob] has adopted [OCI Artifacts][artifacts-repo], generalizing container images as one of many types of artifacts that may be stored in a registry. Other artifact types currently include:
+
+* [Helm Charts][helm-registry]
+* [Singularity][singularity]
+* Car firmware updates, deployed from OCI Artifact registries
+
+## The Need for a Generalized Signing Solution
+
+This document serves as the requirements and constraints of a generalized signing solution. It focuses on the scenarios and needs, and very specifically avoids any reference to other projects or implementations. As our working group forms a consensus on the requirements, the group will then transition to a spec.
+
+## Non-Goals
+
+- Notary v2 does not account for what the content represents or its lineage. Other efforts may attach additional content, and re-sign the super set of content to account for other scenarios. 
+
+## Key Stake Holders & Contributors
+
+As we identify the requirements and constraints, a number of key contributors will be asked to represent their requirements and constraints.
+
+> Please add companies, projects, products that you believe should be included.
+
+* Registry Cloud Operators
+  * [Azure Container Registry (acr)][acr] - Steve Lasker <steve.lasker@microsoft.com> (@stevelasker)
+  * [Amazon Elastic Container Registry (ecr)][ecr] - Omar Paul <omarpaul@amazon.com>
+  * [Docker Hub][docker-hub]
+  * [Google Container Registry (gcr)][gcr]
+  * [GitHub Package Registry (gpr)][gpr]
+  * [Quay][quay]
+  * [IBM Cloud Container Registry (icr)][icr]
+* Registry Vendors, Projects & Products
+  * [Docker Trusted Registry][docker-dtr]
+  * [Harbor][harbor]
+  * [JFrog Artifactory][jfrog]
+* Artifact Types
+  * Container Images - OCI Org
+  * Helm Charts
+  * Singularity
+  * Operator Bundles
+
+## Scenarios
+
+### Scenario #1: Local Build, Sign, Validate
+
+Prior to doing any deployment, a developer can test the: build, sign, validate scenario.
+
+1. Locally build a container image using a non-registry specific `name:tag`, such as:  
+  `$docker build hello-world:1`
+1. Locally sign `hello-world:1` 
```

Comment:
Isn't the statement "locally sign" (and the implication below of the developer having access to signing keys) overly restrictive and unnecessary? What's required here is that a signature is available at point. Where it's generated doesn't have to be defined in this document. This is also related to what @lumjjb suggested under "Open discussions".

Reply:
Hi @lahirurd,
What we're trying to capture here is the scenario is enabled locally, without any dependency on other services. For instance, signing shouldn't be limited to a registry, or a Notary signing server. "local" is contextual. For instance, local could also be on a remote VM or inside a container in a build scenario. The point is it can be done in isolation, with access to keys. 
For instance, a developer may only have access to dev keys they can sign locally on their laptop. The "production" keys are likely more secured, and restricted to roles that have more secure access, such as the identity of the build system.

Reply:
Should we change "local" to "offline" here to try and convey that this scenario is more about being able to sign images outside a registry than about necessarily having keys on laptops?

Reply:
Okay, so I take it that the word "local" was used here to mean "outside a registry". 

My point was that this scenario would be just as valid if the dev used some external signing service to sign the image, without having access to the private key. Something like "2. Generate a signature for `hello-world:1`" might make it clearer that we aren't interested in how/where the signature is computed. My concern is that the two places where we describe signing in detail (this, and [scenario #3 step 3](https://github.com/notaryproject/requirements/pull/1/files#diff-685d3bc1e5cc6093032b54e6b54a81e1R95)) both calling out that the signature is generated locally with local keys, might be interpreted as that being the _only_ way to sign something in Notary v2.

If people feel that it's obviously not the case, please feel free to ignore me :)

Reply:
I agree on the wording "local" seems to be restrictive. Would this be considered a separate scenario? - since it seems like they are targeting separate use-cases - developer vs enterprise.